### ISSUE TITLE
Boolean searching (e.g. OR) not supported anymore

### ISSUE THREAD

**Issue by hannahfrost** [sul-dlss/exhibits#322]
reported to @peetucket by Scott Van Duyne

Hi Peter,
just found this one going over the search.  On the browse page of the Feigenbaum site(https://exhibits.stanford.edu/feigenbaum/browse), the category called "Expert Systems: by the Heuristic Programming Project" returns "0 items" at the moment.  It used to return a couple hundred.

It uses a Title search like this:
Title = "CRYSALIS PUFF OR MYCIN OR HASP"

So I checked the search out live in the search box, and it really pulls up 0 items now, I also tried removing the "OR's", and that also resulted in 0 records, and I then tried selecting from Everything (instead of Title only), that resulted in 2 record returns, which were both long reports that had all four of those terms in them.  

So, the bug: apparently, when you put multiple terms in the search text box, they are all implicitly AND'ed together.  The expected behavior, till now, has been that they should be OR'd.   Also, "AND" "OR", "NOT" keywords and parens "()" used to work as expected (we even documented that Boolean expression feature in one of the "About" pages-- we would like them to keep working) but they do not seem to work anymore.  

Seems like the default config for that search box just got changed up at some point.

**Comment by cbeer** [comment]
Seems likely related to https://issues.apache.org/jira/browse/SOLR-8812 (supposedly fixed, but seemingly still problematic). I'll apply the suggested workaround to that search, but we need to pursue this upstream.

**Comment by peetucket** [comment]
@hannahfrost, did you happen to email Scott with a status update?  I can do this too just to let him to know there is some progress.

**Comment by hannahfrost** [comment]
@peetucket I haven't yet reached out to Scott (I was going to wait until I had heard further from @cbeer about the effectiveness of the workaround). I'll follow up later today. Thank you!

**Comment by peetucket** [comment]
Sounds good thanks. Even with no resolution he'd probably appreciate a quick follow up.

**Comment by snydman** [comment]
@cbeer @ndushay  does this mean boolean searches will not work anymore in any Spotlight exhibit?

**Comment by cbeer** [comment]
Yes (not that anyone does).

**Comment by peetucket** [comment]
Potential workaround suggestions for invetigation:

* add an options in the drop-down menu next to the search box: “Advanced/Boolean”.  This could send specific commands to Solr to allow Boolean operations for just that search, while leaving the other search operations that are required for faceting/etc. unaffected for all other searches.
* investigate negative side effects of configuring solr in such a way to support boolean again (not sure this is possible, but some combination of q.op and mm ?)
* others?

**Comment by peetucket** [comment]
Another comment from Scott about the change in functionality, potential impact on relevancy ranking?  I'm not sure what to make of this, or how relevancy is actually computed: 

There is also a significant impact to non-boolean search users which may not be getting considered, as follows:  I believe the NEW behavior is to intersect the results for each word you enter into the search box.  The OLD behavior was to union the results of each word.  Here is a simple example, which I actually have performed before and after the change happened:

In the OLD and NEW versions of Solr, if you do a "Title text" search for "chess", it will bring up 22 documents; if you do a Title text search for "theory", it will bring up 178 documents.  The difference I want to illustrate is this:  in the NEW version of Solr, if you do a Title text search for "chess theory", and Sort-by-relevance, it will bring up 1 document (only) entitled "Programme Playing Chess - A report on the symposium: Theory and Computing Methods in the Upper Mantle Problem".  Whereas: in the OLD version it would bring up bring up 199 documents (whose titles contained either "chess" or "theory"), and then if you sorted by "Relevance", it would pop the aforementioned document with both words in it to the top of the list. 

This is a big change for the average search user, and in effect a Sort-by-relevance is now somewhat meaningless because, I believe Relevance is being computed as number of words in the query string that are found in each result respectively-- now that all the words are required in every result, all docs must have the highest relevance level!

**Comment by cbeer** [comment]
Presumably by "OLD", he means pre-spotlight? We haven't touched the `mm` setting, and I'd be extremely surprised if the behavior changed across solr versions.

**Comment by peetucket** [comment]
good question, let me clarify with him ... i interpreted as spotlight before and/after the solr upgrade, but he could have been referring to the old site (which is now gone for good)

**Comment by cbeer** [comment]
I've also asked @jvine to maybe chime in about relevancy. We're using the same configuration as searchworks, which we think gives pretty good results (and I'd think a search for "chess theory" should return only the document about chess theory..)

**Comment by jvine** [comment]
Recall is about the presence of the words in the document; relevance is based on which fields the words are in, where they are in relation to each other, density in relation to the size of the document, etc. So we've changed recall, maybe via an unintended configuration change...? Default is OR and for SearchWorks we changed it to AND. When we launched SearchWorks 0.9 with OR, the response was *extremely* negative - relevance was not *at all* enough to satisfy users who felt their result sets were bloated. SearchWorks is a huge index relative to a Spotlight exhibit, and that may make a difference. (OTOH currently the (small) library website search is set to OR and sorted by relevance, and the results are terrible.)

I don't know if it's possible to configure individual Spotlight exhibits differently, but maybe it makes sense for Feigenbaum to revert to OR, given the size, focus, and dedicated users?

**Comment by ndushay** [comment]
SearchWorks is running a much older version of Solr.   I *think* the dispatcher sprint has spun up a sort-of  SearchWorks index on sul-solr  (but only for DOR content) ... perhaps that is a way to see what SearchWorks does on the diff versions of Solr, should that be desired.

I believe SearchWorks *had* to switch to edismax for fixes required for CJK;  perhaps exhibits would be fine with plain dismax.   I'm not sure if the behavior would be different or not.  

Basically, we appear to be in the camp that disagrees with Solr's view that the newest behavior in edismax is "desired."  We are not alone in that camp.  But I doubt we want to pin ourselves to an older version of Solr before this particular edismax change was applied to Solr.
